src/Libraries/CoreNodeModels/DropDown.cs

```diff
@@ -91,45 +90,54 @@ public override NodeInputData InputData
         /// </summary>
         public int SelectedIndex
         {
-            get { return selectedIndex; }
+            get => selectedIndex;
             set
             {
                 //do not allow selected index to
                 //go out of range of the items collection
                 if (value > Items.Count - 1 || value < 0)
                 {
                     selectedIndex = -1;
-                    selectedString = String.Empty;
+                    selectedString = string.Empty;
                 }
                 else
                 {
                     selectedIndex = value;
                     selectedString = GetSelectedStringFromItem(Items.ElementAt(value));
                 }
+
                 RaisePropertyChanged("SelectedIndex");
+                RaisePropertyChanged("SelectedString");
             }
         }
 
-        private string selectedString = String.Empty;
+        private string selectedString = string.Empty;
 
         /// <summary>
         /// String form of current selected item, so derived class
         /// can save customized data
         /// </summary>
         public string SelectedString
         {
-            get { return selectedString; }
+            get => selectedString;
             set
             {
-                if (!string.IsNullOrEmpty(value) && value != selectedString)
+                if (value == selectedString)
+                {
+                    return;
+                }
+
+                if (!string.IsNullOrEmpty(value))
                 {
                     var item = Items.FirstOrDefault(i => GetSelectedStringFromItem(i).Equals(value));
                     // In the case that SelectedString deserialize after SelectedIndex
                     // With a valid item from search, get the index of item and replace the current one. 
                     // If no exact match found, fall back to use the default selectedIndex from deserialization.
-                    selectedIndex = item != null ?
-                        Items.IndexOf(item) :
-                        selectedIndex;
+                    if (item != null)
+                    {
+                        selectedIndex = Items.IndexOf(item);
+                        RaisePropertyChanged("SelectedIndex");
```

Comment:
no `RaisePropertyChanged("SelectedString");` here?

Reply:
it happens down below, line 144 (hidden in the diff)